### PR TITLE
Stream without persister

### PR DESCRIPTION
**Pull request by yigit**
sorry i tried reverting my IDE codestyle settings to project default, which caused more lines to change :/.

we really need spotless :/.

### CODE REVIEW

File: store/src/main/java/com/nytimes/android/external/store4/impl/FetcherController.kt [hunk 1]
```diff
@@ -1,42 +1,53 @@
...
+                            realFetcher(key).map {
+                                StoreResponse.Data(
+                                        it,
+                                        origin = ResponseOrigin.Fetcher
+                                ) as StoreResponse<Input>
+                            }.catch {
+                                emit(StoreResponse.Error(it, origin = ResponseOrigin.Fetcher))
+                            }
                         },
+                        piggybackingDownstream = enablePiggyback,
```
Comment (digitalbuddha) [inline store/src/main/java/com/nytimes/android/external/store4/impl/FetcherController.kt]: when would this be true/false?

File: store/src/main/java/com/nytimes/android/external/store4/impl/multiplex/ChannelManager.kt [hunk 2]
```diff
@@ -65,27 +73,46 @@ class ChannelManager<T>(
         if (this.producer !== producer) {
             return
         }
+        val piggyBacked = mutableListOf<ChannelEntry<T>>()
```
Comment (digitalbuddha) [inline store/src/main/java/com/nytimes/android/external/store4/impl/multiplex/ChannelManager.kt]: does this need to be thread safe?  Should it be a copy on write arraylist instead?

File: store/src/test/java/com/nytimes/android/external/store3/pipeline/FlowStoreTest.kt [hunk 3]
```diff
@@ -410,6 +426,55 @@ class FlowStoreTest(
             )
     }
 
+    @Test
+    fun streamWithoutPersister_cached_nonFlowingFetcher() = testScope.runBlockingTest {
```
Comment (digitalbuddha) [inline store/src/test/java/com/nytimes/android/external/store3/pipeline/FlowStoreTest.kt]: nitpick, can this work as one of those swanky paramterized tests?

File: store/src/main/java/com/nytimes/android/external/store4/impl/FetcherController.kt (hunk 1 above)
Comment (yigit) [reply store/src/main/java/com/nytimes/android/external/store4/impl/FetcherController.kt]: this is true when we want earlier network requests to also receive events sent by later subscriptions. It is only true if there is no source of truth.
Should I just update the code to include this comment? (there is documentation on `enablePiggyback`)

File: store/src/main/java/com/nytimes/android/external/store4/impl/multiplex/ChannelManager.kt (hunk 2 above)
Comment (yigit) [reply store/src/main/java/com/nytimes/android/external/store4/impl/multiplex/ChannelManager.kt]: nope. this is an actor so it runs in order.

File: store/src/test/java/com/nytimes/android/external/store3/pipeline/FlowStoreTest.kt (hunk 3 above)
Comment (yigit) [reply store/src/test/java/com/nytimes/android/external/store3/pipeline/FlowStoreTest.kt]: i can but would need to separate to another test :) i was lazy :D.

File: store/src/main/java/com/nytimes/android/external/store4/impl/FetcherController.kt (hunk 1 above)
Comment (yigit) [reply store/src/main/java/com/nytimes/android/external/store4/impl/FetcherController.kt]: added more documentation.

**Pull request merged**